tools/ratbagd.py

```diff
@@ -289,10 +289,41 @@ def commit(self):
 class RatbagdProfile(_RatbagdDBus):
     """Represents a ratbagd profile."""
 
+    CAP_WRITABLE_NAME = 100
+
     def __init__(self, object_path):
         _RatbagdDBus.__init__(self, "Profile", object_path)
 
     @GObject.Property
+    def capabilities(self):
+        """The capabilities of this profile as an array. Capabilities not
+        present on the device are not in the list. Thus use e.g.
+
+        if RatbagdProfile.CAP_WRITABLE_NAME is in profile.capabilities:
+            do something
+        """
+        return self._get_dbus_property("Capabilities")
+
+    @GObject.property
+    def name(self):
+        """The name of the profile"""
+        return self._get_dbus_property("Name")
+
+    @name.setter
+    def name(self, name):
+        """Set the name of this profile.
+
+        @param name The new name, as string"""
+        if RatbagdProfile.CAP_WRITABLE_NAME not in self.capabilities:
```

Comment:
Just a comment on what I did yesterday (I ended up fixing the car while I wanted to change the lightbulb...).

Just blindly ignoring the capability showed some bad logs (an dbus failure cought as an exception by gobject). The reason is that in ratbagd, when we set a property, we forward the returned value, and if the capability is not there, we forward -1001.

-1001 (error capability) is not known by sd-bus, so it converts it into Failure (Unknown error 1001). I tried to teach sd-bus about 1001 and the other errors we are forwarding over the bus, and it turns out it won't be possible to do it cleanly (we can register new errors in sd-bus, but when parsing the error to forward, sd-bus only rely on a static set of unix errors).

Anyway, I think the easiest solution will be to return 1 for each write to a property and only emit the signal when the property actually changed (like we do). Returning custom errors in dbus calls is not going to be a success.